### PR TITLE
Prepare for Netbox 2.7

### PR DESCRIPTION
**Pull request by cimnine**
Related Issue: N/A

## New Behavior

Netbox 2.7 will support different Redis instances for _caching_ and _webhook queuing_.
Therefore a second Redis server is added for caching.
This second Redis server does not persist it's data.

## Contrast to Current Behavior

Currently only one Redis server serves both purposes, _caching_ and _webhook queuing_.
But the two purposes have different requirements for the Redis server.
E.g. we don't want the webhooks to be lost when "the power goes out". Yet for the caching Redis we don't care about persistence, as the cache can be (and probably has to be) re-populated anyways.

Then the caching Redis is local to one instance of Netbox, whereas the webhook queue Redis is shared by at least two clients, the Netbox web instance and the Netbox rqworker instance.

The updated configuration is made in a backward-compatible way, so that users upgrading from previous versions of Netbox Docker, that don't update the `docker-compose.yml` file, should not have to adjust anything in their setup, although I would recommend anyone to do so.

## Discussion: Benefits and Drawbacks

The drawbacks with two Redis's is of course that it takes a little bit more compute resources and the architecture became a little bit more complex to understand.
But at least from this project's perspective, if a user is getting started, he/she will not have to care about that, because it will just work out of the box, as it did in the past.

## Changes to the Wiki

None. It is technically not even a breaking change.

### CODE REVIEW

File: configuration/configuration.py [hunk 1]
```diff
@@ -172,10 +181,6 @@ def read_secret(secret_name):
 # Time zone (default: UTC)
 TIME_ZONE = os.environ.get('TIME_ZONE', 'UTC')
 
-# The Webhook event backend is disabled by default. Set this to True to enable it. Note that this requires a Redis
-# database be configured and accessible by NetBox (see `REDIS` below).
-WEBHOOKS_ENABLED = os.environ.get('WEBHOOKS_ENABLED', 'False').lower() == 'true'
```
Comment (tobiasge) [inline configuration/configuration.py:177]: Is the Webhook system not optional anymore?

Comment (cimnine) [reply configuration/configuration.py:177]: No. The worker shall — in the future — be used for other background tasks besides webhooks.
See https://github.com/netbox-community/netbox/releases/tag/v2.7-beta1:

> ### WEBHOOKS_ENABLED Configuration Setting Removed ([#3408](https://github.com/netbox-community/netbox/issues/3408))
> 
> As django-rq is now a required library, NetBox assumes that the RQ worker process is running. The installation and upgrade documentation has been updated to reflect this, and the WEBHOOKS_ENABLED configuration parameter is no longer used. Please ensure that both the NetBox WSGI service and the RQ worker process are running on all production installations.

Comment (tobiasge) [reply configuration/configuration.py:177]: Ok, I missed that.

**Comment by TeroPihlaja** [comment]
Any change of getting this merged?

**Comment by cimnine** [comment]
I know that 2.7 of Netbox was released and that Netbox Docker doesn't work out of the box with it right now. We will try to get it out as soon as possible.

But frist, we'll merge #212 to release 0.20.0 of this project. Then we'll begin merging other PRs in the `develop` branch. But there is no ETA since this work is all voluntary and we all have day-jobs as well.

What you can do in the meantime is to build your own Docker Image. Switch to the branch of this PR (`git switch 2.7-prep`) and follow [the guide to extend the base image](https://github.com/netbox-community/netbox-docker/wiki/Build#enhance-the-image). I.e. you create a new Dockerfile `27workaround.Dockerfile`, add something like the following, and run `docker build -f 27workaround.Dockerfile -t netbox:2.7-fixed .`.

```Dockerfile
FROM netboxcommunity/netbox:v2.7

COPY configuration/configuration.py /etc/netbox/config/configuration.py
```

**Comment by cimnine** [comment]
This afternoon I was working hard on getting out scripts and configurations ready for Netbox 2.7.
On my local machine, Netbox 2.7 starts when built from this branch and all the initializers do their job.

That is, there are several breaking changes to the initializers themselves. But I believe that they will be more stable now, as they don't rely on iterating through special lists anymore. This should make our initializers more generic and therefore more robust when future changes occur upstream.

Note: This PR should be rebased onto `develop` once #221 and #222 are merged.

**Comment by cimnine** [comment]
This PR was now rebased onto #221 and #222. The tests for python 3.9 fail, but that doesn't matter at the moment because this is an issue with Netbox and Python 3.9 won't be GA for a while.

Given that, I believe this PR is ready to be merged.

**Review by tobiasge — APPROVED** [review]

**Pull request merged**